dsgrid/config/daily_time_dimension_config.py

```diff
@@ -0,0 +1,175 @@
+)
+from dsgrid.utils.timing import timer_stats_collector, track_timing
+from dsgrid.utils.spark import (
+    get_spark_session,
+)
+from .dimensions import DailyTimeDimensionModel
+from .time_dimension_base_config import TimeDimensionBaseConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class DailyTimeDimensionConfig(TimeDimensionBaseConfig):
+    """Provides an interface to a DailyTimeDimensionModel."""
+
+    @staticmethod
+    def model_class() -> DailyTimeDimensionModel:
+        return DailyTimeDimensionModel
+
+    @track_timing(timer_stats_collector)
+    def check_dataset_time_consistency(self, load_data_df, time_columns) -> None:
+        logger.info("Check DailyTimeDimensionConfig dataset time consistency.")
+        expected_cols = set(self.get_load_data_time_columns())
+        actual_cols = set(time_columns)
+        if expected_cols != actual_cols:
+            msg = (
+                f"DailyTimeDimensionConfig expects columns {expected_cols}, "
+                f"but got {actual_cols}"
+            )
+            raise ValueError(msg)
+
+        # Get the year column name (time_year, weather_year, or model_year)
+        year_col = self.get_year_column()
+        month_col = "month"
+        day_col = "day"
+
+        # Get time ranges
+        time_ranges = self.get_time_ranges()
+        # TODO: need to support validation of multiple time ranges: DSGRID-173
+
+        # Build expected timestamps for all ranges
+        expected_timestamps = []
+        for time_range in time_ranges:
+            expected_timestamps.extend(time_range.list_time_range())
+
+        # Get actual timestamps from dataframe
+        actual_timestamps = [
+            pd.Timestamp(
+                year=x[year_col], month=x[month_col], day=x[day_col], tz=self.get_tzinfo()
+            ).to_pydatetime()
+            for x in load_data_df.select(year_col, month_col, day_col)
+            .distinct()
+            .filter(
+                f"{year_col} IS NOT NULL AND {month_col} IS NOT NULL AND {day_col} IS NOT NULL"
+            )
+            .sort(year_col, month_col, day_col)
+            .collect()
+        ]
+
+        if set(expected_timestamps) != set(actual_timestamps):
+            mismatch = sorted(
+                set(expected_timestamps).symmetric_difference(set(actual_timestamps))
+            )
+            msg = (
+                f"load_data timestamps do not match expected times. "
+                f"mismatch count={len(mismatch)}, first 10={mismatch[:10]}"
+            )
+            raise DSGInvalidDataset(msg)
+
+    def build_time_dataframe(self) -> DataFrame:
+        model_time = self.list_expected_dataset_timestamps()
+        df_time = get_spark_session().createDataFrame(
+            model_time,
+            schema=StructType(
+                [StructField(col, IntegerType(), False) for col in ["year", "month", "day"]]
+            ),
+        )
+        df_time = df_time.withColumnRenamed("year", self.get_year_column())
```

Comment:
@lixiliu, if daily time is using the `weather_year` or `model_year` column at first, should this method continue using those names, or should it just produce a `year` column?